### PR TITLE
[network] Add configuration parameter for using iOS wake-up

### PR DESCRIPTION
**Pull request by lsiepel**
The readme previously stated it did auto detection of iOs device, but to me it seemed hardcoded. I changed this to a configurable option as some IoT devices can't handle the additional port knocking.

- Fixes: https://github.com/openhab/openhab-addons/issues/9575

Test jar 4.2.0: https://1drv.ms/u/s!AnMcxmvEeupwjuUA88rQxWnsCUkymg?e=xXOno9

### CODE REVIEW

File: bundles/org.openhab.binding.network/src/main/java/org/openhab/binding/network/internal/NetworkHandlerConfiguration.java [hunk 1]
```diff
@@ -31,5 +31,6 @@ public class NetworkHandlerConfiguration {
     public Integer retry = 1;
     public Integer refreshInterval = 60000;
     public Integer timeout = 5000;
+    public boolean assumeiOsDevice = true;
```
Comment (radoslavv) [inline bundles/org.openhab.binding.network/src/main/java/org/openhab/binding/network/internal/NetworkHandlerConfiguration.java:34]: I would suggest rather to default this to false for NEW things.
Please see the reasoning here: https://github.com/openhab/openhab-addons/issues/9575#issuecomment-1886812359
But understand for EXISTING devices it is safer to default to true -> if parameter is not defined assume value as true.
Maybe consider to apply [Thing Upgrades](https://www.openhab.org/blog/2023-07-23-openhab-4-0-release.html#thing-upgrades) while Openhab upgrade.

File: bundles/org.openhab.binding.network/src/main/resources/OH-INF/thing/thing-types.xml [hunk 2]
```diff
@@ -62,6 +62,15 @@
 					reduces the load when arping and Wake-on-LAN are used. All interfaces are used when left empty.</description>
 				<advanced>true</advanced>
 			</parameter>
+
+			<parameter name="assumeiOsDevice" type="boolean" required="true">
```
Comment (wborn) [inline bundles/org.openhab.binding.network/src/main/resources/OH-INF/thing/thing-types.xml:66]: I think is more clear to describe the functional behavior:

```suggestion
			<parameter name="wakeUpIOSDevice" type="boolean" required="true">
```

When it becomes configurable the `PROPERTY_IOS_WAKEUP` should also be removed.

File: bundles/org.openhab.binding.network/src/main/resources/OH-INF/thing/thing-types.xml [hunk 3]
```diff
@@ -62,6 +62,15 @@
...
+				<label>Assume iOs Device</label>
+				<default>true</default>
+				<description>Set to true if the device presence detection should be performed for an iOS device like iPhone or
+					iPads. An additional port knock is performed before a ping.</description>
```
Comment (wborn) [inline bundles/org.openhab.binding.network/src/main/resources/OH-INF/thing/thing-types.xml:70]: These texts should also be added to the network.properties file so they can be translated.

File: bundles/org.openhab.binding.network/src/main/resources/OH-INF/thing/thing-types.xml [hunk 4]
```diff
@@ -62,6 +62,15 @@
 					reduces the load when arping and Wake-on-LAN are used. All interfaces are used when left empty.</description>
 				<advanced>true</advanced>
 			</parameter>
+
+			<parameter name="assumeiOsDevice" type="boolean" required="true">
+				<label>Assume iOs Device</label>
```
Comment (jlaur) [inline bundles/org.openhab.binding.network/src/main/resources/OH-INF/thing/thing-types.xml:67]:
```suggestion
				<label>Assume iOS Device</label>
```

File: bundles/org.openhab.binding.network/src/main/java/org/openhab/binding/network/internal/NetworkHandlerConfiguration.java (hunk 1 above)
Comment (lsiepel) [reply bundles/org.openhab.binding.network/src/main/java/org/openhab/binding/network/internal/NetworkHandlerConfiguration.java:34]: No, changing the default behaviour is a breaking change. The Thing upgrade does not affect configuration.

File: bundles/org.openhab.binding.network/src/main/resources/OH-INF/thing/thing-types.xml (hunk 2 above)
Comment (lsiepel) [reply bundles/org.openhab.binding.network/src/main/resources/OH-INF/thing/thing-types.xml:65]: Removed the propertythe property

Comment (jlaur) [reply bundles/org.openhab.binding.network/src/main/resources/OH-INF/thing/thing-types.xml:65]: Did you see the first part of the comment? 😉

Comment (lsiepel) [reply bundles/org.openhab.binding.network/src/main/resources/OH-INF/thing/thing-types.xml:65]: I think `assumeiOsDevice` is a good fit, but i am open to other sugestions. 
The opted  `wakeUpIOSDevice` could be interpreted as a function only limited to iOS devices, but it actually performs an extra port knock, to _all_ devices.

Maybe `useIOSWakeup` ?

Comment (wborn) [reply bundles/org.openhab.binding.network/src/main/resources/OH-INF/thing/thing-types.xml:65]: I think normally a space or - is used (wake up/wake-up) so `useIOSWakeUp` is also fine with me.

Comment (lsiepel) [reply bundles/org.openhab.binding.network/src/main/resources/OH-INF/thing/thing-types.xml:65]: Done

**Review by wborn — APPROVED** [review]
Thanks!

**Pull request merged**